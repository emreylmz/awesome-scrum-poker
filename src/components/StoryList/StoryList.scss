@import "../../styles/colors";
@import "../../styles/metrics";

.story-list {
  display: flex;
  flex: 2;
  flex-direction: column;
  border: $default-border-width solid $primary-color;
  padding: $default-margin;
  border-radius: $default-border-radius;
  margin: $default-margin;

  &__header {
    display: flex;
    flex-direction: row;
    align-content: space-around;
    font-weight: bold;
    padding: $default-margin;
    border-bottom: $default-border-width solid $gray-color;

    &-title {
      display: flex;
      flex: 1;
    }
  }

  &__stories {

  }
}
