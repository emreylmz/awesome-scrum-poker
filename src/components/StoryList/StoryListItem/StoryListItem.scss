@import "../../../styles/colors";
@import "../../../styles/metrics";

.story-list-item {
  display: flex;
  flex-direction: row;
  flex: 1;
  align-content: space-around;

  padding: $default-margin;

  &__name {
    display: flex;
    flex: 1;
  }

  &__point {
    display: flex;
    flex: 1;

  }

  &__status {
    display: flex;
    flex: 1;

  }
}
