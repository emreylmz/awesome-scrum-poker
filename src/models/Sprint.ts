import { Story } from './Story'
import { Voter } from './Voter'

export interface Sprint {
  id: string
  name: string
  numberOfVoters: number
  storyList: Story[]
  activeStory: Story
  inviteLink?: string
  voterList: Voter[]
}
