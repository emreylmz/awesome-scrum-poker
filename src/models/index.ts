export * from './Sprint'
export * from './Story'
export * from './Voter'
