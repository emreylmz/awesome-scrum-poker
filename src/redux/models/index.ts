export * from './reduxAction'
export * from './reduxType'
