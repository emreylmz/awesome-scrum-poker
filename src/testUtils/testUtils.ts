export const emptyTestFunc = (): void => void 0
