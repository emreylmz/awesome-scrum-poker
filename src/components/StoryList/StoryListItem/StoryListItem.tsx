import React from 'react'
import { Render } from '../../../types'
import { classNames } from '../../../utils'
import './StoryListItem.scss'

interface Props {
  title: string
  point?: number
  status: number
  isOdd?: boolean
}

class StoryListItem extends React.PureComponent<Props> {
  getStatusName(status: number): string {
    switch (status) {
      case 0:
        return 'Not Voted'
      case 1:
        return 'Active'
      case 2:
        return 'Voted'
      default:
        return ''
    }
  }
  render(): Render {
    const { title, point, status, isOdd } = this.props
    const className = classNames('story-list-item', isOdd && 'light-background')
    return (
      <div className={className}>
        <div className={'story-list-item__name'}>{title}</div>
        <div className={'story-list-item__point'}>{point}</div>
        <div className={'story-list-item__status'}>{this.getStatusName(status)}</div>
      </div>
    )
  }
}

export type StoryListItemProps = Props
export default StoryListItem
