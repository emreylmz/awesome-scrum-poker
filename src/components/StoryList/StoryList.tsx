import React from 'react'
import { connect } from 'react-redux'
import { Story } from '../../models'
import { RootState } from '../../redux/reducers'
import { Render } from '../../types'
import './StoryList.scss'
import StoryListItem from './StoryListItem/StoryListItem'

interface StateProps {
  stories?: Story[]
}

type Props = StateProps

export class StoryList extends React.PureComponent<Props> {
  render(): Render {
    const { stories } = this.props
    return (
      <div className={'story-list'}>
        <div className={'story-list__header'}>
          <div className={'story-list__header-title'}>Story</div>
          <div className={'story-list__header-title'}>Story Point</div>
          <div className={'story-list__header-title'}>Status</div>
        </div>
        <div className={'story-list__stories'}>
          {stories ? stories.map(story => <StoryListItem key={story.id} {...story} />) : 'No Story'}
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state: RootState): StateProps => {
  return {
    stories: state.SprintReducer.sprint?.storyList
  }
}

export type StoryListProps = Props
export default connect(mapStateToProps)(StoryList)
