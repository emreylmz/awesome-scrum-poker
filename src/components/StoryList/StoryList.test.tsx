import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import { Story } from '../../models'

import '../../testUtils/setupTests'
import { StoryList, StoryListProps } from './StoryList'
import StoryListItem from './StoryListItem/StoryListItem'

type DoneCallback = jest.DoneCallback

const getWrapper = (props?: StoryListProps): ShallowWrapper => {
  return shallow(<StoryList {...props} />)
}

describe('Story List UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it("should have 'No Story' text when there is no story", (done: DoneCallback): void => {
    const wrapper = getWrapper()
    const labelWrapper = wrapper.find('.story-list__stories')
    expect(labelWrapper).toBeDefined()
    expect(labelWrapper.text()).toEqual('No Story')
    done()
  })

  it('should have as many StoryListItem as the number of stories that given with props', (done: DoneCallback): void => {
    const stories: Story[] = [
      {
        id: '1',
        title: 'Story 1',
        status: 1
      },
      {
        id: '2',
        title: 'Story 2',
        status: 0
      },
      {
        id: '3',
        title: 'Story 3',
        status: 0
      }
    ]
    const wrapper = getWrapper({ stories })
    const labelWrapper = wrapper.find(StoryListItem)
    expect(labelWrapper).toBeDefined()
    expect(labelWrapper.length).toEqual(stories.length)
    done()
  })
})
