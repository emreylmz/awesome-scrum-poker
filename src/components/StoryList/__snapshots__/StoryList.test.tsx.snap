// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story List UI should render correctly 1`] = `ShallowWrapper {}`;
